### PR TITLE
Refactored the `verifyDeposit` and `verifyWithdrawal` functions

### PR DESCRIPTION
**Pull request by jalextowle**
# Description

This PR refactors the `verifyDeposit` and `verifyWithdrawal` functions so that they don't need to be reimplemented for each instance.

### CODE REVIEW

**Review by jrhea — APPROVED** [review]
lgtm

File: test/instances/eeth/EETHHyperdrive.t.sol [hunk 1]
```diff
@@ -167,142 +170,14 @@ contract EETHHyperdriveTest is InstanceTest {
 
     /// @dev Fetches the total supply of the base and share tokens.
     function getSupply() internal view override returns (uint256, uint256) {
-        return (POOL.getTotalPooledEther(), EETH.totalShares());
+        return (address(POOL).balance, EETH.totalShares());
     }
 
     /// @dev Fetches the token balance information of an account.
```
Comment (mcclurejt) [inline test/instances/eeth/EETHHyperdrive.t.sol:176]:
```suggestion
    /// @dev Fetches the token balance information of an account.
    /// @return Token balance of the account.
    /// @return Share balance of the account.
```

File: test/utils/InstanceTest.sol [hunk 2]
```diff
@@ -131,6 +131,10 @@ abstract contract InstanceTest is HyperdriveTest {
         /// @dev The equality tolerance for the short at maturity round trip
         ///      with shares test.
         uint256 roundTripShortMaturityWithSharesTolerance;
+        /// @dev The equality tolerance for `verifyDeposit`.
```
Comment (mcclurejt) [inline test/utils/InstanceTest.sol:134]: After reading this description I'm still not super clear on how to grok the tolerance... like what does 2 mean for the tolerance? 2% tolerance? 2 Wei tolerance? Would be nice to have this info here w/o needing to dive into the functions to find out.

**Review by mcclurejt — APPROVED** [review]
LGTM, few nits

**Pull request merged**